client/src/components/LobbySettings.svelte

```diff
@@ -0,0 +1,135 @@
+<script lang="ts">
+  import "@material/typography/mdc-typography.scss";
+  import Dialog, { Header, Title, Content } from "@smui/dialog";
+  import Button, { Label } from "@smui/button";
+  import IconButton from "@smui/icon-button";
+  import Slider from "@smui/slider";
+  import FormField from "@smui/form-field";
+  import { GAME_STATE_DURATIONS_MIN, GAME_STATE_DURATIONS_MAX, type Lobby } from "$lib/firebase/firestore-types/lobby";
+  import { applyLobbySettings } from "$lib/firebase/firebase-functions";
+
+  export let lobby: Lobby;
+  export let lobbyCode: string;
+  let errorMessage: string = "";
+  let showLobbySettings = false;
+  let catfishValue: number = 1;
+  let promptTimerValue = lobby.lobbySettings.promptTime;
+  let chatTimerValue = lobby.lobbySettings.chatTime;
+  let voteTimerValue = lobby.lobbySettings.voteTime;
+
+  const sliderLabels = new Map([
+    ["PROMPT", "Prompt Timer"],
+    ["CHAT", "Chat Timer"],
+    ["VOTE", "Vote Timer"],
+  ]);
+  const minimumTimes = new Map([
+    ["PROMPT", GAME_STATE_DURATIONS_MIN.PROMPT],
+    ["CHAT", GAME_STATE_DURATIONS_MIN.CHAT],
+    ["VOTE", GAME_STATE_DURATIONS_MIN.VOTE],
+  ]);
+  const maximumTimes = new Map([
+    ["PROMPT", GAME_STATE_DURATIONS_MAX.PROMPT],
+    ["CHAT", GAME_STATE_DURATIONS_MAX.CHAT],
+    ["VOTE", GAME_STATE_DURATIONS_MAX.VOTE],
+  ]);
+
+  async function apply(catfish: number, prompt: number, chat: number, vote: number) {
+    try {
+      await applyLobbySettings({
+        code: lobbyCode,
+        catfishNumber: catfish,
+        promptTimer: prompt,
+        chatTimer: chat,
+        voteTimer: vote,
+      });
+    } catch (err) {
+      errorMessage = err instanceof Error ? err.message : String(err);
+    }
+  }
+</script>
+
+<main>
+  <Dialog
+    bind:open={showLobbySettings}
+    fullscreen
+    aria-labelledby="rules-dialog-title"
+    aria-describedby="rules-dialog-content"
+    ><Header>
+      <Title>Lobby Settings</Title>
+      <IconButton action="close" class="material-icons">close</IconButton>
+    </Header>
+    <Content>
+      <div class="settings">
+        <FormField align="end" style={"width:100%"}>
+          <Slider
+            bind:value={catfishValue}
+            min={1}
+            max={3}
+            step={1}
+            discrete
+            input$aria-label="Discrete slider"
+            style={"width:100%;"}
+          />
+          <span slot="label">Catfish Amount</span>
+        </FormField>
+
+        <FormField align="end" style={"width:100%"}>
+          <Slider
```

Comment:
It might be more user-friendly to have minutes:seconds instead of something like 280 seconds since that is harder to gauge immediately

Reply:
If we do use seconds tho these should probably have "seconds" after the number